SCons/Tool/ninja/ninja.xml

```diff
@@ -0,0 +1,325 @@
+            <item>NINJA_SKIP</item>
+            <item>NINJA_CUSTOM_HANDLERS</item>
+-->
+
+        </uses>
+    </tool>
+
+    <builder name="Ninja">
+        <summary>
+            <para>
+                &b-Ninja; is a special builder which
+                adds a target to create a ninja build file.
+                The builder does not require any source files to be specified,
+            </para>
+            <note>
+                <para>This is an experimental feature. To enable it you must use one of the following methods
+                </para>
+
+                <example_commands>
+                    # On the command line
+                    --experimental=ninja
+
+                    # Or in your SConstruct
+                    SetOption('experimental', 'ninja')
+                </example_commands>
+
+                <para>This functionality is subject to change and/or removal without deprecation cycle.</para>
+
+
+                <para>
+                    To use this tool you must install pypi's <ulink url="https://pypi.org/project/ninja/">ninja
+                    package</ulink>.
+                    This can be done via
+                    <command>pip install ninja</command>
+                </para>
+
+            </note>
+
+            <para>
+                If called with no arguments,
+                the builder will default to a target name of
+                <filename>ninja.build</filename>.
+            </para>
+            <para>
+                If called with a single positional argument,
+                &scons; will "deduce" the target name from that source
+                argument, giving it the same name, and then
+                ignore the source.
+                This is the usual way to call the builder if a
+                non-default target name is wanted.
+            </para>
+            <para>
+                If called with either the
+                <parameter>target=</parameter>
+                or <parameter>source=</parameter> keyword arguments,
+                the value of the argument is taken as the target name.
+                If called with both, the
+                <parameter>target=</parameter>
+                value is used and <parameter>source=</parameter> is ignored.
+                If called with multiple sources,
+                the source list will be ignored,
+                since there is no way to deduce what the intent was;
+                in this case the default target name will be used.
+            </para>
+            <para>
+                <emphasis>Available since &scons; 4.2.</emphasis>
+            </para>
+        </summary>
+    </builder>
+
+
+    <cvar name="NINJA_GENERATED_SOURCE_SUFFIXES">
+        <summary>
+            <para>
+                The list of source file suffixes which are generated by SCons build steps.
+                All source files which match these suffixes will be added to the _generated_sources alias in the output
```

Comment:
since this is prelim we can fixup these later - annotation of filenames and the alias name